pkg/connector/telegram.go

```diff
@@ -579,6 +617,94 @@ func (t *TelegramClient) onUpdateNewMessage(ctx context.Context, entities tg.Ent
 	return nil
 }
 
+func (t *TelegramClient) migrateChat(ctx context.Context, oldPortalKey, newPortalKey networkid.PortalKey) error {
+	log := zerolog.Ctx(ctx).With().
+		Str("handler", "migrate_chat").
+		Stringer("old_portal_key", oldPortalKey).
+		Stringer("new_portal_key", newPortalKey).
+		Logger()
+
+	if t.main.Config.AlwaysTombstoneOnSupergroupMigration {
+		return t.migrateChatAndTombstone(ctx, oldPortalKey, newPortalKey)
```

Comment:
You shouldn't need a custom function for this, just create the room for the new portal and continue with the ReIDPortal call